examples/src/main/java/IssueExample.java

```diff
@@ -0,0 +1,139 @@
+    */
+    public static void initParticularNodeAndPingIt() throws PrecheckStatusException, TimeoutException {
+        @Var Map<String, AccountId> networkMap = new HashMap<>();
+        networkMap.put("0.testnet.hedera.com:50211", new AccountId(3));
+
+        Client client = Client.forNetwork(networkMap);
+
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+
+        var network = client.getNetwork();
+        var nodes = new ArrayList<>(network.values());
+
+        var node = nodes.get(0);
+
+        client.setMaxNodeAttempts(1);
+        client.ping(node);
+    }
+
+    // Here is a second example how to ping particular node (fetch specific node from the network map and use only it).
+    public static void initParticularNodeAndPingIt2()
+        throws PrecheckStatusException, TimeoutException, InterruptedException {
+        Client client = Client.forName(HEDERA_NETWORK);
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+
+        var network = client.getNetwork();
+
+        Entry<String, AccountId> firstEntry = network.entrySet().iterator().next();
+        Map<String, AccountId> specificNode = Map.ofEntries(firstEntry);
+
+        client.setNetwork(specificNode);
+
+        client.pingAll();
+    }
+
+    /*
+    From issue:
+    "There are some very big downsides to this approach: for starters, it does not support TLS.
+    That is because no attempt to load the address book is loaded, as that path is not executed.
+    Client provides a setTransportSecurity() flag but this appears to have no effect."
+
+    Comment:
+    "There is no `NodeAddress` instance created in `Network#forNetwork` and no call for `Node#setAddressBookEntry` is made.
+    Hence `certHash` is null and we're getting an exception in example below."
+     */
+    public static void TLSWithoutNodeAddresses() throws PrecheckStatusException, TimeoutException, InterruptedException {
+        @Var Map<String, AccountId> networkMap = new HashMap<>();
+        networkMap.put("0.testnet.hedera.com:50211", new AccountId(3));
+
+        Client client = Client.forNetwork(networkMap).setTransportSecurity(true).setVerifyCertificates(true);
+
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+
+        var network = client.getNetwork();
+        var nodes = new ArrayList<>(network.values());
+
+        var node = nodes.get(0);
+
+        client.setMaxNodeAttempts(1);
+        client.ping(node);
+    }
+
+    // However here we can set network map with only specific nodes populated from address book and it would work.
+    public static void TLSWithoutNodeAddresses2() throws PrecheckStatusException, TimeoutException, InterruptedException {
+        Client client = Client.forName(HEDERA_NETWORK).setTransportSecurity(true).setVerifyCertificates(true);
+        client.setOperator(OPERATOR_ID, OPERATOR_KEY);
+
+        var network = client.getNetwork();
+
+        Entry<String, AccountId> firstEntry = network.entrySet().iterator().next();
```

Comment:
Is this confirmed to be a TLS node and this passes?

I don't know order that is returned by this set, and I believe order of set is not guaranteed? But if it returns first element mirrornode returns it [will be non tls endpoint](https://testnet.mirrornode.hedera.com/api/v1/network/nodes). Would be good to know. If this works, this can be considered a work-around

Reply:
Hey @dalvizu, so:
- I can confirm the order of the set is not guaranteed;
- When setting `.setTransportSecurity(true).setVerifyCertificates(true)` -- `Client#getNetwork` returns `:50212` endpoints, which are TLS ones;

(You can confirm it in the debugger by checking the `network` variable 🙂)

Also, the mirror node shouldn’t be part of those networks. You can choose whichever account you want to test from the `network`.